test/functional/feature_onion.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test bitcoind with 'onion'-tagged peers.
+
+Tested outbound and inbound peers.
+"""
+
+import os
+
+from test_framework.p2p import P2PInterface
+from test_framework.socks5 import Socks5Configuration, Socks5Server
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    PORT_MIN,
+    PORT_RANGE,
+    assert_equal,
+)
+
+
+class OnionTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 2
+        self.setup_clean_chain = True
+
+    def setup_network(self):
+        port_offset = PORT_MIN + 2 * PORT_RANGE + (os.getpid() % 1000)
+        self.bind_onion = '127.0.0.1:{}'.format(port_offset + 1000)
+
+        self.proxy_conf = Socks5Configuration()
+        self.proxy_conf.addr = ('127.0.0.1', port_offset + 2000)
+        self.proxy_conf.unauth = True
+        self.proxy = Socks5Server(self.proxy_conf)
+        self.proxy.start()
+
+        args = [
+            ['-bind={}=onion'.format(self.bind_onion), '-onion=%s:%i' % (self.proxy_conf.addr)],
+            [],
+            ]
+        self.add_nodes(self.num_nodes, extra_args=args)
+        self.start_nodes()
+
+    def run_test(self):
+        self.log.info("Checking a node without any peers...")
+        network_info = self.nodes[0].getnetworkinfo()
+        assert_equal(network_info["connections"], 0)
+        assert_equal(network_info["connections_in"], 0)
+        assert_equal(network_info["connections_out"], 0)
+        assert_equal(network_info["connections_onion_only"], False)
```

Comment:
This repetitious code could potentially be refactored into a small helper function `get_connection_info()`.

Reply:
Thanks! [Reworked](https://github.com/bitcoin/bitcoin/pull/20172#issuecomment-712479264).